### PR TITLE
Added possibility to add custom simple buttons

### PR DESCRIPTION
**Pull request by bolelamx**

### CODE REVIEW

File: src/javascript/app/ng-wig/ngWigToolbarProvider.js [hunk 1]
```diff
@@ -1,23 +1,39 @@
...
-    list1: {title: 'Unordered List', command: 'insertunorderedlist', styleClass: 'nw-button--unordered-list'},
-    list2: {title: 'Ordered List', command: 'insertorderedlist', styleClass: 'nw-button--ordered-list'},
-    bold: {title: 'Bold', command: 'bold', styleClass: 'nw-button--bold'},
-    italic: {title: 'Italic', command: 'italic', styleClass: 'nw-button--italic'},
-    link: {title: 'Link', command: 'createlink', styleClass: 'nw-button--link'}
+    list1: {title: 'Unordered List', command: 'insertunorderedlist', styleClass: 'nw-button--unordered-list', isSimple: true},
+    list2: {title: 'Ordered List', command: 'insertorderedlist', styleClass: 'nw-button--ordered-list', isSimple: true},
+    bold: {title: 'Bold', command: 'bold', styleClass: 'nw-button--bold', isSimple: true},
+    italic: {title: 'Italic', command: 'italic', styleClass: 'nw-button--italic', isSimple: true},
+    link: {title: 'Link', command: 'createlink', styleClass: 'nw-button--link', isSimple: true}
```
Comment (besuhoff) [inline src/javascript/app/ng-wig/ngWigToolbarProvider.js]: what about isCustom? By default it is false and doesn't require specific data when you add a simple button

File: src/javascript/app/ng-wig/ngWigToolbarProvider.js [hunk 2]
```diff
@@ -1,23 +1,39 @@
...
+    if(!Array.isArray(buttons)) {
+      throw 'Argument "buttons" should be an array';
+    }
 
-      defaultButtonsList = buttons;
+    defaultButtonsList = buttons;
   }
 
+  this.addSimpleButton = function (name, options) {
+    if (typeof name !== "string") {
```
Comment (stevermeister) [inline src/javascript/app/ng-wig/ngWigToolbarProvider.js]: there is a helper `angular.isString`

File: src/javascript/app/ng-wig/ngWigToolbarProvider.js [hunk 3]
```diff
@@ -1,23 +1,39 @@
...
+    }
+
+    var lowerCaseName = name.toLowerCase(),
+        defaultOptions = {
+          title: lowerCaseName[0].toUpperCase() + lowerCaseName.slice(1),
+          command: lowerCaseName,
+          styleClass: 'nw-button--' + lowerCaseName
+        };
+
+    buttonLibrary[name] = angular.extend({},defaultOptions, options, {isSimple: true});
```
Comment (besuhoff) [inline src/javascript/app/ng-wig/ngWigToolbarProvider.js]: ({}, defaultOptions)

File: src/javascript/app/ng-wig/ngWigToolbarProvider.js (hunk 1 above)
Comment (stevermeister) [inline src/javascript/app/ng-wig/ngWigToolbarProvider.js]: +1
it's better to put here param like "complex"(or better name) and it's false by default

File: src/javascript/app/ng-wig/ngWigToolbarProvider.js [hunk 4]
```diff
@@ -1,23 +1,39 @@
...
   }
 
+  this.addSimpleButton = function (name, options) {
+    if (typeof name !== "string") {
+      throw 'Argument "name" should be a string';
+    }
+
+    var lowerCaseName = name.toLowerCase(),
+        defaultOptions = {
+          title: lowerCaseName[0].toUpperCase() + lowerCaseName.slice(1),
```
Comment (stevermeister) [inline src/javascript/app/ng-wig/ngWigToolbarProvider.js]: too much magic. 
it's better to separate name/key and the title

File: src/javascript/app/ng-wig/ngWigToolbarProvider.js [hunk 5]
```diff
@@ -1,23 +1,39 @@
...
+          command: lowerCaseName,
```
Comment (stevermeister) [inline src/javascript/app/ng-wig/ngWigToolbarProvider.js]: ...and command as well

Let's make API of your module more intuitive.

**Comment by bolelamx** [comment]
New icons
screenshot from 2015-09-16 16 55 33

File: bower.json [hunk 6]
```diff
@@ -2,7 +2,8 @@
   "name": "ngWig",
   "version": "1.1.10",
   "devDependencies": {
-    "angular": "~1.2.16"
+    "angular": "~1.4.5",
```
Comment (stevermeister) [inline bower.json]: without ~

File: src/javascript/app/ng-wig/ngWigToolbarProvider.js [hunk 7]
```diff
@@ -1,23 +1,41 @@
...
+  this.addSimpleButton = function (name, title, command, styleClass) {
+    if (!name || !angular.isString(name)) {
+      throw 'Argument "name" is required and should be a string';
+    }
+
+    if (!title || !angular.isString(title)) {
+      throw 'Argument "title" is required and should be a string';
+    }
+
+    if (!command || !angular.isString(command)) {
```
Comment (stevermeister) [inline src/javascript/app/ng-wig/ngWigToolbarProvider.js]: to many checks.
`!name || !title || !command` should be enough

**Comment by stevermeister** [comment]
:+1:

**Pull request merged**